dartagnan/src/main/java/com/dat3m/dartagnan/analysis/graphRefinement/graphs/eventGraph/stat/ProgramOrderGraph.java

```diff
@@ -0,0 +1,60 @@
+package com.dat3m.dartagnan.analysis.graphRefinement.graphs.eventGraph.stat;
+
+import com.dat3m.dartagnan.analysis.graphRefinement.util.EdgeDirection;
+import com.dat3m.dartagnan.program.Thread;
+import com.dat3m.dartagnan.verification.model.Edge;
+import com.dat3m.dartagnan.verification.model.EventData;
+import com.dat3m.dartagnan.verification.model.ExecutionModel;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class ProgramOrderGraph extends StaticEventGraph {
+
+    private Map<Thread, List<EventData>> threadEventsMap;
+
+    @Override
+    public boolean contains(Edge edge) {
+        return edge.isForwardEdge();
```

Comment:
The definition of ```isForwardEdge``` checks that both events belong to the same thread. IMO, the notion of ```forward``` is only defined within a thread. We could define it via the ```cId``` only, but I think this might be more misleading.